app-admin/terraform/terraform-0.12.5.ebuild

```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit golang-vcs-snapshot
+
+DESCRIPTION="A tool for building, changing, and combining infrastructure safely"
+HOMEPAGE="https://www.terraform.io/"
+
+EGO_PN="github.com/hashicorp/${PN}"
+SRC_URI="https://github.com/hashicorp/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MPL-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+DEPEND=">=dev-lang/go-1.11.0"
+
+DOCS=( src/${EGO_PN}/{README,CHANGELOG}.md )
+
+src_compile() {
+	cd "src/${EGO_PN}" || die
+	GOPATH="${S}" GOCACHE="${T}/go-cache" go build \
+		-v -work -o "${S}/${PN}" ./ || die
+}
+
+src_install() {
+	dobin terraform
+
+	einstalldocs
+}
+
+pkg_postinst() {
+	elog "If you would like to install shell completions please run:"
+	elog "    terraform -install-autocomplete"
```

Comment:
Hmm, why exactly can't the ebuild install them the regular way?

Reply:
so all that command does is add `complete -C /usr/bin/terraform terraform` to bashrc. I don't think the binary actually drops a file to disk.

Reply:
Instead of playing with bashrc, you should dump a proper completion file via bash-completion-r1 so it's autoloaded for all users.

Reply:
I'm having difficulty getting it to drop a completion file, I'll keep trying though

Reply:
Write it by hand. I think you basically need the same thing as it writes into bashrc, named `terraform`.